File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    let str = s.toLowerCase();
    let n = str.replace(/[^a-z0-9]/g , "");
    let left = 0;
    let right = n.length-1;

    while(left < right){
        if(n[left] != n[right]){
            return false;
        }
        left++;
        right--;
    }
    return true;
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String str = s.toLowerCase();
        String ch = str.replaceAll("[^a-z0-9]" , "");
        int left = 0;
        int right = ch.length()-1;
        while(left < right){
            if(ch.charAt(left) != ch.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}